purchase_request/models/purchase_request.py

```diff
@@ -0,0 +1,299 @@
+        comodel_name="res.users",
+        required=True,
+        copy=False,
+        tracking=True,
+        default=_get_default_requested_by,
+        index=True,
+    )
+    assigned_to = fields.Many2one(
+        comodel_name="res.users",
+        string="Approver",
+        tracking=True,
+        domain=lambda self: [
+            (
+                "groups_id",
+                "in",
+                self.env.ref("purchase_request.group_purchase_request_manager").id,
+            )
+        ],
+        index=True,
+    )
+    description = fields.Text()
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        required=True,
+        default=_company_get,
+        tracking=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name="purchase.request.line",
+        inverse_name="request_id",
+        string="Products to Purchase",
+        readonly=False,
+        copy=True,
+        tracking=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        related="line_ids.product_id",
+        string="Product",
+        readonly=True,
+    )
+    state = fields.Selection(
+        selection=_STATES,
+        string="Status",
+        index=True,
+        tracking=True,
+        required=True,
+        copy=False,
+        default="draft",
+    )
+    is_editable = fields.Boolean(compute="_compute_is_editable", readonly=True)
+    to_approve_allowed = fields.Boolean(compute="_compute_to_approve_allowed")
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        string="Picking Type",
+        required=True,
+        default=_default_picking_type,
+    )
+    group_id = fields.Many2one(
+        comodel_name="procurement.group",
+        string="Procurement Group",
+        copy=False,
+        index=True,
+    )
+    line_count = fields.Integer(
+        string="Purchase Request Line count",
+        compute="_compute_line_count",
+        readonly=True,
+    )
+    move_count = fields.Integer(
+        string="Stock Move count", compute="_compute_move_count", readonly=True
+    )
+    purchase_count = fields.Integer(
+        string="Purchases count", compute="_compute_purchase_count", readonly=True
+    )
+    currency_id = fields.Many2one(related="company_id.currency_id", readonly=True)
+    estimated_cost = fields.Monetary(
+        compute="_compute_estimated_cost",
+        string="Total Estimated Cost",
+        store=True,
+    )
+
+    @api.depends("line_ids", "line_ids.estimated_cost")
+    def _compute_estimated_cost(self):
+        for rec in self:
+            rec.estimated_cost = sum(rec.line_ids.mapped("estimated_cost"))
+
+    @api.depends("line_ids")
+    def _compute_purchase_count(self):
+        for rec in self:
+            rec.purchase_count = len(rec.mapped("line_ids.purchase_lines.order_id"))
+
+    def action_view_purchase_order(self):
+        action = self.env.ref("purchase.purchase_rfq").sudo().read()[0]
```

Comment:
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("purchase.purchase_rfq")
```
use this instead sudo+read all over places

Reply:
@i-vyshnevska thanks irina, all done